sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -822,6 +821,8 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   def table(tableName: String): DataFrame = {
     assertNoSpecifiedSchema("table")
+    for ((k, v) <- this.extraOptions)
+      sparkSession.conf.set(k, v)
```

Comment:
We need to clearly define the behavior here.

I think the options don't make sense if we read temp view, view, or v1 tables in the session catalog. For v1 table, they are created with `CREATE TABLE  ... USING ... OPTIONS ...`, so the options are already there and it's confusing to overwrite them per scan.

We do need to keep the scan options when reading v2 tables, but I don't think session conf is the right channel. I think we should put the options in `UnresolvedRelation` and apply these options when we resolve `UnresolvedRelation` to v2 relations.

Reply:
@cloud-fan Thanks for your comment. 
I took a look at the code. I think I need to change the following:

<img width="896" alt="Screen Shot 2020-08-25 at 6 03 49 PM" src="https://user-images.githubusercontent.com/13592258/91242868-5aa37e80-e6fd-11ea-8498-2554d5c9367a.png">


<img width="900" alt="Screen Shot 2020-08-25 at 6 02 38 PM" src="https://user-images.githubusercontent.com/13592258/91242806-38116580-e6fd-11ea-8057-91ee3c3ef58e.png">

<img width="859" alt="Screen Shot 2020-08-25 at 6 02 50 PM" src="https://user-images.githubusercontent.com/13592258/91242813-3a73bf80-e6fd-11ea-9b67-5196f4597f73.png">

I need to modify lots of files because of the above changes. Could you please check if this looks OK to you before I continue? Thank you very much!

Reply:
1. We don't need to change `SparkSession.table`. We can create `UnresolvedRelation` in `DataFrameReader.load` directly.
2. This is options to scan the v2 table, not v2 table properties. We don't need to change `TableCatalog.loadTable`, but pass the scan options to `Table.newScanBuilder`. `DataSourceV2Relation` already takes the scan options, we just need to create `DataSourceV2Relation` with scan options in `ResolveTables` and `ResolveRelations`.

Reply:
Currently, we already pass options from `DataFrameReader` to `DataSourceV2Relation` in `load` API. `DataFrameReader.table` misses this part.